happi/client.py

```diff
@@ -535,3 +535,35 @@ def find_config():
         # If found nothing
         raise EnvironmentError("No happi configuration file found. "
                                "Check HAPPI_CFG.")
+
+    def choices_for_field(self, field):
+        """
+        List all choices for a given field
+
+        Parameters
+        ----------
+        field : string
+            search field to list all possible choices for
+            i.e 'beamline', 'name', 'z', 'prefix', etc.
+
+        Raises
+        ------
+        SearchError
+            If no devices in the database have an entry for the given field
+
+        Returns
+        -------
+        field_choices : list
+            list of choices for a given field that are in the database
+        """
+        field_choices = set()
+        for dev in self.all_devices:
+            try:  # Want to ignore error if 'dev' doesn't have 'field'
+                choice = getattr(dev, field)
+                if choice not in field_choices:
```

Comment:
Do you need this `if` anymore? Think `Set` handles this for you.

Reply:
I most certainly do not. Changed in a89f330